# Methods

## Collection 

The data that we are going to use in these analyses mostly came from the MalariaGEN Vector Observatory [cite website]. The MalariaGEN Vector Observatory is a global collaboration that covers several different projects with different foci, areas of expertise and area of collection under the umbrella of the Pan-African Mosquito Control Association (PAMCA) [cite website]. For these analyses, the data that is used comes mostly from the ANOSPP [cite ANOSPP] and the Anopheles gambiae genomic surveillance project [cite website], as well as the now complete Ag1000G project [cite paper] that forms the foundation of the Vector Observatory. 

One of the main sample sets that we are going to analyse is a set of samples that were collected at high altitude using nets tied to a balloon [cite paper]. This sample set contains 23 samples (22 An. coluzzii and 1 An. gambiae) that were collected in 4 locations (Dallowere, Markabougou, Siguima and Thierola) from 2013 to 2015 during the rainy season (from June to September; the only An. gambiae sample was collected in October). Splitting these samples into cohorts depending on collection location, year and/or month of collection is going to yield sample counts lower than what would be necessary for most analyses. For this reason, the samples coming from this aerial collection study will be either considered as a unified cohort (sometimes excluding the obviously much different unique An. gambiae sample), grouped with ground samples that were collected in the same location and/or around the same date or used for analyses at the sample level.

Another key sample set for our analyses comes from the same group. [find more about collection methods]. The samples were collected from 2 locations: Dallowere and Sokourani; and across 4 years: 2012-2015. At each location, samples were collected during the late dry season (March-May), early wet season (June-July) and late wet season/early dry season (September-November). Due to issues during sequencing, in particular with the older samples, the cohort size for these samples is lower than would be ideal and samples will thus have to be gathered from different time points to get cohorts of sufficient size for some analyses. Table ? shows the distribution of collected samples across locations and dates of collection. We can observe that only An. coluzzii were collected in Soukourani while some An. gambiae s.s. and some An. arabiensis were collected from Dallowere, though in smaller numbers than An. coluzzii. As the same collection methods were used in both locations, this might show that conditions in Soukourani provide An. coluzzii mosquitoes with an advantage against other taxa that doesn't exist in Dallowere or, at least, that the conditions impact the behaviour of the mosquitoes such that An. gambiae s.s. and An. arabiensis samples would not be collected.

We also use a historical data set collected in southern Mali during the late wet season (August-September) in 2004 [cite paper]. These samples came from 6 villages: Bancoumana, Down, Fanzana, Kela, Moribobougou and N'Gabakoro. All samples were collected by spray-catching. Overall, similar numbers of An. gambiae s.s. and An. coluzzii were collected as well as a small number of An. arabiensis. Almost all An. gambiae s.s. samples came from 2 villages (Bancoumana and Kela) where no samples from other taxa were collected. In general, once again, the cohort sizes at each location (with a few exceptions) are quite small and samples will generally be grouped by taxon in our analyses.

Another set of samples came from a study of the anopheles population around the Guinea/Mali border [cite paper]. We are mostly interested in the samples from Mali in this paper. They were collected in October 2012 in southern Mali using a combination of human-landing capture, indoor manual aspirator or pyrethroid spray catch, and larval capture. Two samples were collected in two villages: Takan (arid savannah) and Toumani Oulena (humid savannah). In both villages, one taxon was found to be predominant (An. coluzzii in Takan and An. gambiae s.s. in Toumani Oulena) which might indicate that the different taxa are differently adapted to the conditions in the two villages. In our analyses, we will generally group these samples by taxa without consideration of the location of collection.

Many samples, coming both from Mali and Burkina Faso, came through our collaboration with the Target Malaria project [cite paper]. In Mali, samples were collected in 2014 and 2015 during the late wet season or the early dry season (August to November) in 4 locations: Kababougou, Ouassorola, Sogolombougou and Tieneguebougou. Samples were collected by human landing catch and pyrethrum spray collection. No An. arabiensis was collected in any of these locations and, except in Tieneguebougou, more An. gambiae s.s. were collected than An. coluzzii. In Burkina Faso, samples were collected from 2014 to 2017 across the whole year (though not every month every year) in 3 locations: Bana, Pala and Souroukoudinga by human landing catch, pyrethrum spray collection or aspiration. Almost all An. arabiensis were collected in Pala. In all locations, except for Pala, more An. coluzzii were collected than An. gambiae s.s.. Additionally, some more recent samples from these 3 villages were collected during the late wet season and early dry season (August to October). Same collection methods? More An. arabiensis were collected from outside Pala and the trend of collecting more An. coluzzii than An. gambiae s.s. (and vice-versa in Pala) seems to be confirmed (though the numbers are close in Souroukoudinga). Because there are so many different time points, most cohorts are fairly small and will thus mostly be grouped either by location and taxon or by year and taxon depending on the analyses.

A few more samples were collected by indoor spraying in 2004 in Monomtenga in Central Burkina Faso [cite paper]. Samples were collected during the wet season (June to August) and were all gambiae. There are, however, only 13 such samples and they are thus usually going to be grouped together as a single cohort.

In addition to these data coming from the Ag1000G and the Vector Observatory projects, we will also use some data that was collected and sequenced wholly outside of the scope of these collaborations but was made public on ENA/NCBI [link to websites]. Some of this data comes from work by the University of California Malaria Initiative [cite Lanzaro papers] that contains data from Mali. Samples were collected from 2002 to 2012 as larvae or adults. Because each cohort contains a small number of samples they will either be grouped together or be grouped with other samples when they are used for comparisons. We will also use samples collected in 2004 in Mali and Burkina Faso by MichaÃ«l Fontaine et al. [cite Fontaine]. The collections took place in Bana, Pala and Souroukoudinga in Burkina Faso and Kela in Mali which are all locations we already have data. Once more, because the cohort sizes are small these data will generally be grouped with other samples from the same (or neighbouring) location(s). We also make use of data from adults and larvae (then reared to adults) collected by Crawford et al. [cite Crawford] (Goundry was only found among larvae, worth mentioning?) and data from larvae reared to adults collected by Jacob Tennessen et al. [cite Tennessen]. New subtaxa (An. goundry and An. tengrela, respectively) were introduced in these papers and these can be used as comparison. Other analyses [cite method paper] tend to show that these two subtaxa may be the same. These papers also introduced some An. arabiensis and An. coluzzii that can be compared to our data.

## Sequencing, alignments and quality control

Library preparations and sequencing for the samples from the Ag1000G and Vector Observatory projects took place at the Wellcome Sanger Institute. Samples were sequenced using either the Illumina HiSeq 2000 platform or the Illumina HiSeq X platform. The samples were sequenced with a target coverage of 30X. 

Sequence reads were aligned to the An. gambiae reference genome AgamP4. BWA version 0.7.15. was used for alignments and GATK version 3.7-0 RealignerTargetCreator and IndelRealignerwere used for indel realignments. Specifications and implementations for these tasks can be found on GitHub [add link to the repos]. More details can be found in [cite paper].

Samples went through a quality control step. Samples were considered to have low coverage if their median coverage depth (across the genome) was below 10X or if less than 50% of the genome had more than 1X coverage depth. Samples were checked for cross-contamination following the method outlined in [cite Jun et al.]. Samples were also excluded if the ratio of the modal coverage for X and for 3R (representing the autosomes) diverged from 0.5x (male) or 1x (female). Because some samples were sequenced more than once, technical replicates were excluded. Finally, population outliers were identified using PCA (computed using scikit-allele version 12.0.0 [cite or link]). Once more, [cite paper] contains more details on this process.


## SNPs, CNVs and haplotypes

GATK version 3.7-0 UnifiedGenotyper was used to call single nucleotide polymorphisms (SNP). Our focus will mostly be in analysing SNPs around known target-site insecticide resistance loci in order to compare profiles between taxa, locations and time points.

In order to compute copy number variations (CNV), we follow the method outlined in [cite Lucas]. CNVs are estimated by looking at the number of reads in non-overlapping windows using pysam [cite or link]. A Hidden-Markov Model was then used to ascertain actual values. Similarly to the SNPs, the analysis of the CNVs will mostly center around insecticide resistance genes but will focus on metabolic resistance instead. 

In order to compute the haplotypes from the genotypes, we built a pipeline [cite resource paper and/or link the pipeline repo] that uses WhatsHap [cite WhatsHap] and ShapeIT [cite ShapeIT]. In order to estimate relatedness, we use the haplotypes to generate haplotype networks and haplotype clusterings that group together shared haplotypes and represents how they are connected to each other. We particularly generate such graphs for regions known to be under selection (such as known insecticide resistance loci) where only a few different background exist and thus it is possible to classify samples into different groups connected by gene flow.

## Diversity statistics

We will use nucleotide diversity (\pi) [cite paper?], Watterson's theta [cite paper?] and Tajima's D [cite paper?])  to estimate genetic diversity. These statistics are computed using most of chromosome arm 3L (15,000,000-41,000,000) using only 4-fold degenerate coding sites. We chose to use 3L to avoid known introgressions and inversions on other chromosome arms and thus have a picture of diversity under neutral evolution. Both ends of the chromosome arms are excluded as the centromere and the telomere tend to contain highly repetitive regions for which sequencing is alignment is thus harder. We then compare these statistics between cohorts in order to detect similarities and differences.  We also compute Fst [cite paper] and average genetic distance between cohorts to compare populations depending on taxa, locations or time points. FST is also computed using the same region of the genome.

## Locator

Locator [cite Locator paper] is a program that attempts to infer the location of collection of individuals from their genome. It uses a deep-network approach to do so. We used the tool as is, at least in part to compare its runs on the newest available data from the Ag1000G project (phase 3) to the runs by the tool's creators that ran it on data from phase 2. However, because the data contains samples from the same location belonging to different taxa (e.g., Anopheles gambiae s.s. and Anopheles coluzzii from Dallowere in Mali), there are samples that would have the same coordinates but be much more different genetically than samples from across the African continent. This causes the problem of locating samples to be a very difficult problem. In order to try to avoid this issue, we modified the code of Locator to work more as a classifier trying to infer a country and a taxon instead of coordinates. This causes other problems (e.g., is it true that samples from the Kenyan shore of Lake Victoria are more similar to samples on the oceanic coast than they are to samples from the Ugandan shore of Lake Victoria?) that may have to be addressed in the future.

## Doubleton-sharing

In order to focus on close relationships between individuals, we decided to look at rare variants and how they are shared between populations. More specifically, we focused on doubleton-sharing. The main idea is to take a cohort of mosquitoes, generally composed of two populations that we want to compare. We then filter to keep only positions of the genome where exactly two copies of a given allele are found in the cohort in two different individuals (i.e., there are two copies of the allele and no individual is homozygote for this allele). In order to compare populations, we then count the number of sites where the allele is shared by two individuals belonging to the same population and how many sites are shared by two individuals belonging to two different populations. It is expected that populations that share more rare variants are more closely related than populations that do not share as many rare variants. Similarly, it would be expected that a sample that shares more rare variants with a population would be closer genetically to that population than to another population with which it shares less rare variants. The number of doubletons is highly dependent on the cohort size, as adding an extra sample can introduce new doubletons (if there is a site where the new sample is heterozygous for a singleton of the original cohort) as well as remove some (if there is a site where the new sample is heterozygous or homozygous for a doubleton of the original cohort). The size of the cohorts thus needs to be taken into consideration when comparing analyses.

## IBD

Another approach to estimate relatedness that we used is to look at the length of shared identical regions between individuals. Each mosquito inherits a copy of each chromosome from each parent (modulo recombination during meiosis I) and thus shares long regions of identical haplotypes with each of its parents. Because of different recombination events, siblings share shorter regions, and so on as mosquitoes-relatedness decreases. These shared regions are said to be identical by descent (IBD). These regions are broken down by recombination from generation to generation and thus become shorter. In practice, it is impossible to know if two mosquitoes have identical regions of the genome because they are related (i.e., due to IBD) or if it is due to other mechanisms (e.g., regions under strong selection are likely to be more conserved than regions that are not and thus to stay identical longer). It is only possible to observe identity by state (IBS), that is that two mosquitoes have a region of shared haplotype. Following results in human genomics [cite IBD papers], and using the method described in [cite Sanjay], we compute IBS by looking at breakpoints between shared regions. We have two different types of break points: we say that we encounter a breakpoint of type 1 if there is any difference between the two pairs of alleles at a given site; we say that we encounter a breakpoint of type 2 if, in addition, both samples are homozygous for different alleles at that site. 